content/vault/v1.21.x/content/docs/deploy/run-as-zcx-cluster.mdx

```diff
@@ -0,0 +1,317 @@
+---
+
+# Run Vault on IBM z/OS Container Extensions (zCX)
+
+This guide walks through deploying a fully secured 3-node HashiCorp Vault Enterprise cluster
+on IBM z/OS Container Extensions (zCX). The setup uses three independent zCX instances, 
+each running Vault with its own unique IP address. A Layer-4 HAProxy load balancer sits 
+in front to distribute traffic, and end-to-end TLS encryption is applied across all Vault nodes 
+and the load balancer to ensure secure communication throughout the cluster.
+
+![Vault zCX cluster deployment](/img/run-as-zcx-cluster.png)
+
+## Step 1: Container image deployment
+
+1. Pull the official Vault Enterprise container image on all three zCX nodes.
+
+  ```shell-session
+  $ docker pull hashicorp/vault-enterprise
+  ```
+
+1. Verify image integrity.
+
+  ```shell-session
+  $ docker images | grep vault-enterprise
+  ```
+
+## Step 2: Persistent volume creation
+
+Create Docker volumes for configuration and data persistence on each node.
+
+1. Create volume for Vault data storage (Raft backend, audit logs).
+
+  ```shell-session
+  $ docker volume create vault-data
+  ```
+
+1. Create volume for Vault configuration files.
+
+  ```shell-session
+  $ docker volume create vault-config
+  ```
+
+1. Verify volume creation.
+
+  ```shell-session
+  $ docker volume ls | grep vault
+  ```
+
+Volume mount points:
+
+- **`vault-data`**: `/vault/data` (stores Raft snapshots, encrypted storage backend)
+- **`vault-config`**: `/vault/config.d` (HCL configuration files)
+
+## Step 4: Vault configuration file setup
+
+Create Vault configuration on each node with node-specific parameters.
+
+1. Access config volume.
+
+  ```shell-session
+  $ docker run --rm -it -v vault-config:/vault alpine sh
+  ```
+
+1. Create configuration file for Leader and follower vault.
+
+  ```hcl
+  ui = true
+  disable_mlock = true
+
+  # --- Listener Configuration ---
+  listener "tcp" {
+    address              = "0.0.0.0:8200"
+    tls_disable          = 0
+    tls_cert_file        = "/vault/vault.pem"
+    tls_key_file         = "/vault/vault.key"
+    tls_client_ca_file   = "/vault/ca.pem"
+  }
+
+  # --- Raft Storage (Leader Node) ---
+  storage "raft" {
+    path    = "/vault/data"
+    node_id = "<LEADER_NODE_ID>"  # e.g., "node1"
+
+    # No retry_join block needed for leader
+  }
+
+  # --- Cluster Networking ---
+  api_addr     = "https://<LEADER_IP>:8200"
+  cluster_addr = "https://<LEADER_IP>:8201"
+  ```
+
+  Exposes Vault on port 8200 for API and UI traffic.
+
+   - **`tls_disable`** = 0 → TLS is enabled (mandatory for secure cluster communication).
+
+### Certificates:
+
+- **`vault.pem`** → TLS certificate
+
+- **`vault.key`** → Private key
+
+- **`ca.pem`** → CA used to validate mutual TLS between nodes
+
+This ensures encrypted traffic between Vault clients, HAProxy, and other nodes.
+
+Similarly create configuration for follower Vault nodes.
```

Comment:
```suggestion
1. Create a configuration file for Vault follower nodes:. For example, the
   following configuration file exposes Vault on port 8200 for API and UI
   traffic, sets `tls_disable` to `0` to force TLS communication within the
   cluster, and sets the leader information so that followers know how to join
   the cluster:
```